<?xml version="1.0"?>
<launch>
  <!-- By default we do not overwrite the URDF. Change the following to true to change the default behavior -->
  <!-- The name of the parameter under which the URDF is loaded -->
  <arg name="robot_description" default="robot_description"/>
  <arg name="trajopt_description" default="trajopt_description"/>
  <arg name="plotting" default="true"/>
  <arg name="write_to_file" default="false"/>
  <arg name="rviz" default="true"/>
  <arg name="ifopt" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="testing" default="false"/>

  <!-- Load universal robot description format (URDF) -->
  <param name="$(arg trajopt_description)" textfile="$(find tesseract_ros_examples)/config/glass_upright_plan.json"/>

  <!-- Load universal robot description format (URDF) -->
  <param name="$(arg robot_description)" command="$(find xacro)/xacro --inorder '$(find tesseract_ros_examples)/urdf/lbr_iiwa_14_r820.xacro'" />

  <!-- The semantic description that corresponds to the URDF -->
  <param name="$(arg robot_description)_semantic" textfile="$(find tesseract_ros_examples)/config/kuka_lbr_iiwa_14_r820.srdf" />

  <!-- Load updated joint limits (override information from URDF) -->
  <group ns="$(arg robot_description)_planning">
    <rosparam command="load" file="$(find tesseract_ros_examples)/config/joint_limits.yaml"/>
  </group>

  <!-- Load default settings for kinematics; these settings are overridden by settings in a node's namespace -->
  <group ns="$(arg robot_description)_kinematics">
    <rosparam command="load" file="$(find tesseract_ros_examples)/config/kinematics.yaml"/>
  </group>

  <group unless="$(arg testing)">
    <!-- If needed, broadcast static tf for robot root -->
    <node pkg="tf" type="static_transform_publisher" name="virtual_joint_broadcaster_0" args="0 0 0 0 0 0 world base_link 100" />

    <node pkg="tesseract_ros_examples" type="tesseract_ros_examples_glass_upright_example_node" name="tesseract_ros_examples_glass_upright_example_node" output="screen">
      <param name="plotting" type="bool" value="$(arg plotting)"/>
      <param name="write_to_file" type="bool" value="$(arg write_to_file)"/>
      <param name="rviz" type="bool" value="$(arg rviz)"/>
      <param name="ifopt" type="bool" value="$(arg ifopt)"/>
      <param name="debug" type="bool" value="$(arg debug)"/>
    </node>

    <!-- Launch visualization -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="tesseract_ros_examples_glass_upright_example_rviz"
        args="-d $(find tesseract_ros_examples)/config/examples.rviz" />
  </group>

</launch>
